frontend/packages/console-shared/src/components/dropdown/dropdown-with-switch/DropdownWithSwitchGroups.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable @typescript-eslint/ban-ts-ignore */
+import * as React from 'react';
+import { Divider, MenuGroup, MenuItem, MenuList } from '@patternfly/react-core';
+
+const DropdownWithSwitchGroups: React.FC<DropdownWithSwitchGroupsProps> = ({
+  options,
+  selectedKey,
+}) => {
+  return options.length === 0 ? null : (
+    <>
+      {options.map(({ items, label }) =>
+        items.length === 0 ? (
+          // PatternFly expects Menu to contain a MenuList with a MenuItem
+          // see https://github.com/patternfly/patternfly-react/issues/7365
+          // hack to workaround this bug by adding a hidden MenuList
+          <MenuList className="pf-u-display-none" key="hidden-placeholder-menu">
```

Comment:
It's there because of the bug in PatternFly.

Reply:
I'm not sure if I am following -- you're returning `null` if `options` are empty. How is this any different? Is it because `null` is not liked by PF? We can filter those out after the map... eg `.filter((v) => !!v)`

Reply:
I guess if the `options.length === 1 && options[0].items.length === 0` you will have an issue?

If that's the case, be more declarative and filter all empty `.items` out before you map.

Reply:
Did you read [the upstream bug I filed](https://github.com/patternfly/patternfly-react/issues/7365)?  The issue has to do with focus handling in PatternFly...

Reply:
Apologies, I missed it in your comment -- was wrapped up looking at the code & why you needed the ignore and skipped right past it. On reply I was lost in something else. 👍  Definitely a funky workaround for a bug in PF.

FWIW, you can do `<MenuItem translate="no" />` to get around your `ignore` issue